src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreRT.cs

```diff
@@ -185,10 +185,163 @@ public static unsafe void StructureToPtr(object structure, IntPtr ptr, bool fDel
+                    return new System.PlatformNotSupportedException();
+                case HResults.COR_E_RANK:
+                    return new System.RankException();
+                case HResults.COR_E_REFLECTIONTYPELOAD:
+                    return new System.MissingMethodException();
+                case HResults.COR_E_RUNTIMEWRAPPED:
+                    return new System.MissingMethodException();
+                case HResults.COR_E_SECURITY:
+                    return new System.Security.SecurityException();
+                case HResults.COR_E_SERIALIZATION:
+                    return new System.Runtime.Serialization.SerializationException();
+                case HResults.COR_E_STACKOVERFLOW:
+                    return new System.StackOverflowException();
+                case HResults.COR_E_SYNCHRONIZATIONLOCK:
+                    return new System.Threading.SynchronizationLockException();
+                case HResults.COR_E_SYSTEM:
+                    return new System.SystemException();
+                case HResults.COR_E_TARGET:
+                    return new System.Reflection.TargetException();
+                case HResults.COR_E_TARGETINVOCATION:
+                    return new System.MissingMethodException();
+                case HResults.COR_E_TARGETPARAMCOUNT:
+                    return new System.Reflection.TargetParameterCountException();
+                case HResults.COR_E_THREADABORTED:
+                    return new System.Threading.ThreadAbortException();
+                case HResults.COR_E_THREADINTERRUPTED:
+                    return new System.Threading.ThreadInterruptedException();
+                case HResults.COR_E_THREADSTART:
+                    return new System.Threading.ThreadStartException();
+                case HResults.COR_E_THREADSTATE:
+                    return new System.Threading.ThreadStateException();
+                case HResults.COR_E_TYPEACCESS:
+                    return new System.TypeAccessException();
+                case HResults.COR_E_TYPEINITIALIZATION:
+                    return new System.TypeInitializationException("");
+                case HResults.COR_E_TYPELOAD:
+                    return new System.TypeLoadException();
+                case HResults.COR_E_TYPEUNLOADED:
+                    return new System.TypeUnloadedException();
+                case HResults.COR_E_UNAUTHORIZEDACCESS:
+                    return new System.UnauthorizedAccessException();
+                //case HResults.COR_E_UNSUPPORTEDFORMAT:
+                case HResults.COR_E_VERIFICATION:
+                    return new System.Security.VerificationException();
+                //case HResults.E_INVALIDARG:
+                case HResults.E_NOTIMPL:
+                    return new System.NotImplementedException();
+                //case HResults.E_POINTER:
+                case HResults.RO_E_CLOSED:
+                    return new System.ObjectDisposedException("");
```

Comment:
Is this sufficient to make all tests pass? I see that Mono has the mapping for a bunch more HRESULTs, and there are tests validating it (e.g. src\libraries\System.Runtime\tests\System\IO\FileLoadException.InteropTests.cs).

It may be nice to move the Mono impl into the shared partition.

Reply:
Yeah, I didn't rerun the tests with this. I foolishly deleted the entries for which we didn't have the HRESULT definition. We also need to copy the HRESULT definitions and keep those all.